docs/markdown/snippets/dep_as_static.md

```diff
@@ -0,0 +1,13 @@
+## `dep.as_static()` and `dep.as_shared()` functions
+
+Internal dependencies can now choose between static and shared version
+of a `both_libraries` linked library. This allows to use the same
+dependency objects for building either a static or a shared library from
+the same hierarchy of dependecies.
+
+`internal_depencency` now have `.as_static()` and `.as_shared()` methods, to
+convert it to a dependency that prefer either version of the linked targets.
+
+Add `prefer_static` keyword to `internal_dependency()`, to set the default
+prefered version. Otherwise, uses the `prefer_static` option to select the
+prefered version.
```

Comment:
Not sure `prefer_static` is needed. I fear it could cause inconsistent behaviour across projects. Some will decide to prefer static, others don't, and in the end you never know if you're getting a shared or static library when using default_library=both. I prefer keeping it consistent and always prefer shared, and let the user of that dependency choose to use as_static() explicitly.

What do you think?

Reply:
Note that if we always keep shared by default, then as_shared() is not needed.